dist/src/test/java/io/camunda/application/CamundaDockerIT.java

```diff
@@ -0,0 +1,128 @@
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+
+@EnabledIfSystemProperty(named = "camunda.docker.test.enabled", matches = "true")
+public class CamundaDockerIT {
+
+  private static final int SERVER_PORT = 8080;
+  private static final int MANAGEMENT_PORT = 9600;
+  private static final int GATEWAY_GRPC_PORT = 26500;
+  private static final int ELASTICSEARCH_PORT = 9200;
+
+  private static final String CAMUNDA_NETWORK_ALIAS = "camunda";
+  private static final String ELASTICSEARCH_NETWORK_ALIAS = "elasticsearch";
+
+  private static final String CAMUNDA_TEST_DOCKER_IMAGE =
+      System.getProperty("camunda.docker.test.image", "camunda/camunda:SNAPSHOT");
+  private static final String ELASTICSEARCH_DOCKER_IMAGE =
+      System.getProperty(
+          "camunda.docker.test.elasticsearch.image",
+          "docker.elastic.co/elasticsearch/elasticsearch:8.14.1");
+
+  private GenericContainer camundaContainer;
+  private ElasticsearchContainer elasticsearchContainer;
+
+  @Test
+  public void testStartCamundaDocker() throws Exception {
+    elasticsearchContainer =
+        new ElasticsearchContainer(ELASTICSEARCH_DOCKER_IMAGE)
+            .withNetwork(Network.SHARED)
+            .withNetworkAliases(ELASTICSEARCH_NETWORK_ALIAS)
+            .withEnv("xpack.security.enabled", "false")
+            .withExposedPorts(ELASTICSEARCH_PORT);
+    elasticsearchContainer.start();
+    camundaContainer =
+        new GenericContainer<>(CAMUNDA_TEST_DOCKER_IMAGE)
+            .withExposedPorts(SERVER_PORT, MANAGEMENT_PORT, GATEWAY_GRPC_PORT)
+            .withNetwork(Network.SHARED)
+            .withNetworkAliases(CAMUNDA_NETWORK_ALIAS)
+            .waitingFor(
+                new HttpWaitStrategy()
+                    .forPort(MANAGEMENT_PORT)
+                    .forPath("/actuator/health")
+                    .withReadTimeout(Duration.ofSeconds(120)))
+            .withStartupTimeout(Duration.ofSeconds(300))
+            .withEnv(
+                "ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_CLASSNAME",
+                "io.camunda.zeebe.exporter.ElasticsearchExporter")
+            .withEnv("ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_URL", elasticsearchUrl())
+            .withEnv("ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_BULK_SIZE", "1")
+            .withEnv("CAMUNDA_TASKLIST_ELASTICSEARCH_URL", elasticsearchUrl())
+            .withEnv("CAMUNDA_TASKLIST_ZEEBEELASTICSEARCH_URL", elasticsearchUrl())
+            .withEnv("CAMUNDA_TASKLIST_ZEEBE_GATEWAYADDRESS", gatewayAddress())
+            .withEnv("CAMUNDA_TASKLIST_ZEEBE_RESTADDRESS", httpUrl())
+            .withEnv("CAMUNDA_OPERATE_ELASTICSEARCH_URL", elasticsearchUrl())
+            .withEnv("CAMUNDA_OPERATE_ZEEBEELASTICSEARCH_URL", elasticsearchUrl())
+            .withEnv("CAMUNDA_OPERATE_ZEEBE_GATEWAYADDRESS", gatewayAddress());
+    try {
```

Comment:
🔧 could we move some parts out of the test into a set up method or fields, to make it more clearer what the test is actually trying to do?

Reply:
fixed in https://github.com/camunda/camunda/pull/19623/commits/ecd6e586376a0d2feedd0bc87cf4fbc31594a4b8